protocol/contracts/profit/src/contract.rs

```diff
@@ -51,15 +48,23 @@ pub fn instantiate(
     )
     .grant_to(&msg.timealarms)?;
 
-    State::new(Config::new(
-        msg.cadence_hours,
-        msg.treasury,
-        OracleRef::try_from(msg.oracle, deps.querier)?,
-        TimeAlarmsRef::new(msg.timealarms, deps.querier)?,
-    ))
-    .store(deps.storage)?;
+    let init_state = State::start(
+        Config::new(
+            msg.cadence_hours,
+            msg.treasury,
+            OracleRef::try_from(msg.oracle, deps.querier)?,
+            TimeAlarmsRef::new(msg.timealarms, deps.querier)?,
+        ),
+        msg.dex,
+    );
+
+    let response = MessageResponse::messages_only(init_state.enter());
+
+    let state: State = init_state.into();
```

Comment:
have you tried to inline this variable?

Reply:
Yes, I've tried, but the into() cannot define the type unless it is assigned to it.

Reply:
I have a solution - `State::start` to return `(State, Batch)`. That would:
1. let us keep the visibility intact
2. eliminate the need to `into` the state here